compiler/GHC/Rename/Expr.hs

```diff
@@ -696,7 +697,7 @@ postProcessStmtsForApplicativeDo ctxt stmts
        -- -XApplicativeDo is on.  Also strip out the FreeVars attached
        -- to each Stmt body.
          ado_is_on <- xoptM LangExt.ApplicativeDo
-       ; let is_do_expr | DoExpr <- ctxt = True
+       ; let is_do_expr | DoExpr{} <- ctxt = True
```

Comment:
I think we need the `m` here and to pass it to `rearrangeForApplicativeDo` , since `rearrangeForApplicativeDo` calls 
```
  (return_name, _) <- lookupSyntaxName returnMName
  (pure_name, _)   <- lookupSyntaxName pureAName
```
See here (https://github.com/tweag/ghc/blob/f9c3eb3cbfbafda69d573a3e22ee9268213f4153/compiler/GHC/Rename/Expr.hs#L1532), and that's where we'll need to make sure it's getting the right return and pure.

Reply:
I'm pointing at this comment from #470. Doesn't look like we need to do something about it in advance.